src/lib/operators/projection.cpp

```diff
@@ -59,6 +61,29 @@ void Projection::_create_column(boost::hana::basic_type<T> type, const std::shar
     // Explicitly pass T{} because in some cases it won't initialize otherwise
     auto values = pmr_concurrent_vector<T>(row_count, T{});
 
+    column = std::make_shared<ValueColumn<T>>(std::move(values), std::move(null_values));
+  } else if (expression->type() == ExpressionType::Select) {
+    auto chunk = expression->table()->get_chunk(ChunkID{0});
+    auto base_column = chunk->get_column(ColumnID{0});
+
+    // the subquery result table can only contain exactly one column with one row
+    // since we checked for this at subquery execution time we can make some assumptions here
+    T subselect_value{};
+    if (auto value_column = std::dynamic_pointer_cast<const ValueColumn<T>>(base_column)) {
+      subselect_value = value_column->get(ChunkOffset{0});
+    } else if (auto dictionary_column = std::dynamic_pointer_cast<const DictionaryColumn<T>>(base_column)) {
```

Comment:
At first, we always used `Table::getValue`, which made the code more concise.
However this leads to performace warnings. In this case though, I don't think it is a performance problem.

Reply:
I don’t dynamically casting a column and then calling a method to access the value is faster than simply calling a virtual method. I’d go for the more concise version.

Reply:
Right. Let's just use getValue for every type of column, add a `PerformanceWarningDisabler` and add a comment saying that this is ok because we only access a single value anyway.

Reply:
Done